portfolio/src/main/java/com/google/sps/servlets/ContactMeFormSubmissionServlet.java

```diff
@@ -1,16 +1,44 @@
 package com.google.sps.servlets;
 
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.KeyFactory;
 import java.io.*;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.*;
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
 
 
 @WebServlet("/submitForm")
 public class ContactMeFormSubmissionServlet extends HttpServlet{
 
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-      
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
+      // get form information
+      String name = Jsoup.clean(request.getParameter("name"), Whitelist.none());
+      String email = Jsoup.clean(request.getParameter("email"), Whitelist.none());
+      String reason = Jsoup.clean(request.getParameter("reason"), Whitelist.none());
```

Comment:
Optional: In newer versions of `jsoup`, [`Whitelist`](https://jsoup.org/apidocs/org/jsoup/safety/Whitelist.html) is deprecated in favor of `Safelist`. Feel free to use `Safelist` if you prefer (though I think it requires upgrading the version of jSoup in your pom.xml to at least 1.15.1 so I'm not too sure it works).